lib/bpf/bpf.go

```diff
@@ -255,6 +259,10 @@ func (s *Service) CloseSession(ctx *SessionContext) error {
 	// Stop watching for events from this PID.
 	s.watch.Remove(cgroupID)
 
+	if err := s.open.endSession(cgroupID); err != nil {
+		return trace.Wrap(err)
```

Comment:
Now when I think about it, I think we should reverse the order here. First, we should remove the cgroup and then remove the cgroup number from the kernel. Otherwise, we have a race condition here (some disk events may be skipped as we remove the cgroup ID from the BPF first before the cgroup is removed).